core/ArchiveProcessor.php

```diff
@@ -351,7 +351,12 @@ protected function aggregateDataTableRecord($name, $columnsAggregationOperation
             // see https://github.com/piwik/piwik/issues/4377
             $self = $this;
             $dataTable->filter(function ($table) use ($self, $columnsToRenameAfterAggregation) {
-                $self->renameColumnsAfterAggregation($table, $columnsToRenameAfterAggregation);
+
+                /** @var \Piwik\Period $period */
+                $period = $table->getMetadata('period');
+                if (!$period || $period->getLabel() === 'day') {
+                    $self->renameColumnsAfterAggregation($table, $columnsToRenameAfterAggregation);
```

Comment:
This makes archiving and range dates a lot faster. Imagine we archive a week, then we will rename all columns of each 7 day archives. Afterwards we know the columns will be replaced in a week archive.

When generating month archives, which uses mostly week archives, we do not have to replace those columns for the week archives again since we can be sure they were already replaced. Same when aggregating year and range archives. 

This can save up 10% or more when aggregating Month, Year and Range archives.